engine/src/main/java/org/terasology/utilities/procedural/PerlinNoise.java

```diff
@@ -71,9 +83,9 @@ public PerlinNoise(long seed) {
      */
     @Override
     public float noise(float posX, float posY, float posZ) {
-        int xInt = (int) TeraMath.fastFloor(posX) & 255;
-        int yInt = (int) TeraMath.fastFloor(posY) & 255;
-        int zInt = (int) TeraMath.fastFloor(posZ) & 255;
+        int xInt = Math.floorMod((int) TeraMath.fastFloor(posX), permCount);
```

Comment:
Why do we `floor` two times here, or what exactly does `floorMod` do?

Reply:
`%` doesn't handle negative numbers well, for those cases `floorMod` is needed and `fastFloor` converts a float to int while always reducing it casting to `int` will give issues- `(int)-7.65 == -7` while `-8` is needed here

Reply:
ok in the simplex implementation I found the better version to use here `floorToInt`, swapping it in